add redux-observable ping pong epic

jlederman opened the pull request:
Click the 'Start Ping' button, the `PING` action is triggered and `isPinging` is set to true. Two seconds later, PingPongEpic triggers the `PONG` action and the reducer sets `isPinging` to false.

jlederman committed:
Add redux-observable ping pong epic

jlederman committed:
Show the value of isPinging as a string in the UI

rlaneve commented:
@jlederman Can this be done using `redux-logic`? Might still be implemented using RxJs/observables, but the hope was to incorporate `redux-logic` to handle the easier scenarios (AJAX stuff, primarily), so that we don't have to go full-complicated on every async scenario like we did when we used `redux-saga` instead of `redux-thunk`. The ping/pong isn't a complicated example, but totally fine to treat it as such.

I also pushed a couple of commits, which caused a couple conflicts (store and routes), but should be easy to resolve those. Our changes are independent, they just both added a new store and new route.

jlederman committed:
Include the redux-logic package

rlaneve commented:
@jlederman Any updates on this?

jlederman committed:
Attempt to implement redux-logic

The 'PONG' action doesn't fire after a few seconds as expected. 'PING' works fine.

jlederman committed:
Working redux-logic implementation with TypeScript

jlederman committed:
Refactor redux-logic ping pong fetch

Attemping to log the result of a Promise without letting
the Promise resolve will always result in a "pending" response

Not including the rest of the existing state when the reducer
returns causes unexpected keys in the store

jlederman commented:
@rlaneve Take a look at the two most recent commits in this PR regarding redux-logic when you get a chance

rlaneve commented:
@jlederman Cool. So we can maintain typing for AJAX responses. Can we do the same for actions coming into redux-logic? That was the other piece, right? (Sorry...been too long since we last discussed this.)

IOW, can we recreate a flow mimicking our real-world use case where an `UPDATE_MODEL` action in the equipment list needs to trigger a fetch to the back-end for item info, and pass along the new model value (a property of the payload of the `UPDATE_MODEL` action). Having redux-logic (or observable, or whatever) listening for a `PING` is nice, but the code responding to that action doesn't depend on any details of the action itself, beyond its type. I'd like to see a strongly-typed action handled instead, where the payload is carrying typed information.

jlederman commented:
I'm closing this seven-year-old PR.
